### ISSUE TITLE
Completed Spring Task Pod Cleanup

### ISSUE THREAD

**Issue by rcmontoy** [spring-cloud/spring-cloud-deployer-kubernetes#131]
Hello,

I'm currently using the Kubernetes implementation of the SCDF deployer to launch short lived tasks and had a few questions:

1. I was looking for some guidance on how to manage the 'cleanup' of completed Spring tasks. The [docs](http://docs.spring.io/spring-cloud-dataflow-server-kubernetes/docs/1.1.2.RELEASE/reference/htmlsingle/#_tasks) indicate that tasks are launched as bare pods and therefore need to be manually cleaned up. This can lead to a large number of 'Completed' pods when using `kubectl get pods -a`. How have others managed to address this issue? periodically running `kubectl delete <pod>`, k8s garbage collection, etc.?

2. If there isn't a suggested approach, are there any plans in the roadmap for tasks to 'cleanup' after themselves (i.e. they do not appear in `kubectl get pods -a` after completion)?

3. What was the motivation from converting task deployments to bare pods instead of k8s jobs? Completed in this PR https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/pull/55

Any help would be greatly appreciated. Thank you!

Environmental info
```
spring-cloud-dataflow-server-kubernetes: 1.1.2.RELEASE
Kubernetes 1.5.2
```

**Comment by trisberg** [comment]
Hi,

We added a feature to cleanup task execution resources (for Kubernetes that's the completed pod) using the `task execution cleanup --id <id>` where id is the id you get from a `task execution list`. This feature is available in the 1.2.0 version which is currently available as 1.2.0.RC1 and will be released next week.

We have had some discussions of adding this to the SCDF UI, but we haven't implemented that yet (see https://github.com/spring-cloud/spring-cloud-dataflow/issues/1056)

We don't have any plans for other types of cleanup or garbage collection currently but we are open to suggestions that would work well across all types of supported runtime platforms.

The reason we went with bare pods is that the approach using jobs wasn't working well for tasks that failed. There was no way to prevent the task from being repeatedly re-launched after failure which makes little sense for our uses case. If the job fails once it is likely failing again, especially if there is a configuration issue for the task.

Hope this helps.

**Comment by rcmontoy** [comment]
Hi @trisberg, I appreciate the quick response.

I think we may end up implementing some sort of periodic cleanup of completed task execution resources using the new API once we upgrade to 1.2.0.

The reason I asked about the switch to bare pods was because k8s jobs provide [cleanup](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#job-termination-and-cleanup), however, it makes little sense to restart failing tasks since k8s jobs are [designed to relaunch until completion](https://github.com/kubernetes/kubernetes/issues/20255).

Thanks again and I appreciate the work on this project.
